pkg/kubelet/lifecycle/handlers.go

```diff
@@ -105,6 +107,31 @@ func resolvePort(portReference intstr.IntOrString, container *v1.Container) (int
 	return -1, fmt.Errorf("couldn't find port: %v in %v", portReference, container)
 }
 
+// formatURL formats a URL from args.
+func formatURL(scheme string, host string, port int, path string) *url.URL {
+	// split path into path and query
```

Comment:
This looks a lot better from a compatibility perspective. It probably deserves a comment about why we're doing this this way. You might also rename the `path` arg to `rawPathAndQuery` or something to make it clear it should already be encoded.